hibernate-core/src/main/java/org/hibernate/type/descriptor/java/DurationJavaType.java

```diff
@@ -120,11 +112,17 @@ public Duration fromString(CharSequence string) {
 		}
 
 		if (value instanceof BigDecimal) {
-			return fromDecimal( value );
+			final BigDecimal decimal = (BigDecimal) value;
+			final BigDecimal[] bigDecimals = decimal.divideAndRemainder( BILLION );
+			return Duration.ofSeconds(
+					bigDecimals[0].longValueExact(),
+					bigDecimals[1].longValueExact()
+			);
```

Comment:
@beikov why didn't you go back and look at the original version of this code, which had the very informative comment that you deleted:

```
			return Duration.ofSeconds(
					secondsAndNanos[0].longValueExact(),
					// use intValue() not intValueExact() here, because
					// the database will sometimes produce garbage digits
					// in a floating point multiplication, and we would
					// get an unwanted ArithmeticException
					secondsAndNanos[1].intValue()
			);
```

Reply:
Thanks for the fix for this. I don't know what the comment means exactly, an example of when anything but `intValue()` fails would be helpful, but I merged your change regardless.